### PR TITLE
[rush] Clean up some unhelpful incremental build logging messages.

### PR DESCRIPTION
**Pull request by Claudiazhaoya**
This PR aims to fix the issue: #1733

### CODE REVIEW

**Comment by iclanton** [comment]
an you give an example of the output you're removing?

File: apps/rush-lib/src/logic/taskRunner/ProjectTask.ts [hunk 1]
```diff
@@ -31,13 +31,11 @@ export interface IProjectTaskOptions {
 function _areShallowEqual(object1: JsonObject, object2: JsonObject, writer: ITaskWriter): boolean {
   for (const n in object1) {
     if (!(n in object2) || object1[n] !== object2[n]) {
-      writer.writeLine(`Found mismatch: "${n}": "${object1[n]}" !== "${object2[n]}"`);
       return false;
     }
   }
   for (const n in object2) {
     if (!(n in object1)) {
-      writer.writeLine(`Found new prop in obj2: "${n}" value="${object2[n]}"`);
```
Comment (sachinjoseph) [inline apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: Change these to print only in debug mode rather than removing these completely?

Comment (Claudiazhaoya) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: That is a good idea!

**Comment by Claudiazhaoya** [comment]
> Can you give an example of the output you're removing?
Here is an example of the output that I just removed.
```
FAILURE (2)
================================
applicaton_a (2 minutes 59.3 seconds)
>>> applicaton_a
react-scripts build
Creating an optimized production build...
Failed to compile.
/some/full/file/path.js
Cannot find file './file_that_doesnt_exist.css' in '/some/full/file/path.js'.

application_b (2 minutes 59.3 seconds)
>>> application_b
Found new prop in obj2: "full/file/CMakeLists.txt" value="e0fee1adf795c84eec4735f039503eb18d9c35cc"
react-scripts build
Creating an optimized production build...
Failed to compile.
/some/full/file/path.js
Cannot find file './file_that_doesnt_exist.css' in '/some/full/file/path.js'.
================================

[applicaton_a] Returned error code: 1
[application_b] Returned error code: 1
```

File: apps/rush-lib/src/logic/taskRunner/ProjectTask.ts (hunk 1 above)
Comment (octogonz) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: What is "debug mode"?

Comment (octogonz) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: BTW my interpretation is that this `_areShallowEqual()` is simply a utility for copying two JSON objects, which returns `true` or `false`.  These messages are printed when it returns `false` but that doesn't indicate any problem, any more than inappropriately returning `true` could be a problem.  Thus these messages are really only useful to a person who is debugging Rush.

If we don't want to delete them entirely (which I was fine with), maybe they could just be code comments. (?)

File: common/changes/@microsoft/rush/zhas-delete-redundant-rush-rebuild-output_2020-03-27-19-31.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Delete redundant rush rebuild output",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/zhas-delete-redundant-rush-rebuild-output_2020-03-27-19-31.json]:
```suggestion
      "comment": "Remove some misleading log output for \"rush build\" (GitHub #1733)",
```

File: apps/rush-lib/src/logic/taskRunner/ProjectTask.ts (hunk 1 above)
Comment (Claudiazhaoya) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: I was thinking the debug mode Sachin refers might be something like 
```
@debug
writer.writeLine(`Found new prop in obj2: "${n}" value="${object2[n]}"`);
```
Java has such feature and I'm looking into if Typescript has it or not. But I haven't found any.  @sachinjoseph Any suggestion here? Or we can just comment them as a reference for debugging :)

Comment (iclanton) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: That isn't a feature in TypeScript.

Let's just comment them out.

Comment (sachinjoseph) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: What I meant was "rush --debug build"

Comment (octogonz) [reply apps/rush-lib/src/logic/taskRunner/ProjectTask.ts]: We really should provide a way for Rush to produce diagnostic logging output and optionally redirect it to a file.  This "logging" should be a separate channel from the "console" output that interacts with users.  (The concepts are fundamentally different, e.g. logging has verbosity levels and sometimes categories, whereas console output has colors and cursor movements.)

As far as I know, it's out of scope for this PR, though.  We should open a separate GitHub issue.

**Pull request merged**